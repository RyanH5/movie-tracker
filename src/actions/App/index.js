export const userSignOut = (id) => ({
  type: 'USER_SIGN_OUT',
  id
});