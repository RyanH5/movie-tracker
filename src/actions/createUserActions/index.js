export const userCreateSuccess = (id) => ({
  type: 'CREATE_USER_SUCCESS',
  id
});

export const requiredIncomplete = (bool) => ({
  type: 'INCOMPLETE_ENTRIES',
  fieldsIncomplete: bool
});

export const creationDenied = (bool) => ({
  type: 'EMAIL_NOT_AVAILABLE',
  emailBeingUsed: bool
});

export const createNewUser = (name, email, password) => {
  return async (dispatch) => {
    dispatch(requiredIncomplete(false));
    dispatch(creationDenied(false));
    if (!name || !email || !password) {
      return dispatch(requiredIncomplete(true));
    }
    try {
      const url = 'http://localhost:3000/api/users/new';
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify(
          {
            name,
            email,
            password
          }
        ),
        headers: {
          'Content-Type': 'application/json'
        }
      });
      const userCreated = await response.json();
      dispatch(userCreateSuccess(userCreated.id));
    } catch (e) {
      dispatch(creationDenied(true));
    }
  };
};
