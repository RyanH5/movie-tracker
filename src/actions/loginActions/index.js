export const fetchIsLoading = (bool) => ({
  type: 'FETCH_LOADING',
  isLoading: bool
});

export const fetchErrored = (bool) => ({
  type: 'FETCH_ERRORED',
  isErrored: bool
});

export const userFetchSuccess = (user) => ({
  type: 'USER_FETCH_SUCCESS',
  user
});

export const fetchDatabase = (url, email, password) => {
  return async (dispatch) => {
    try {
      dispatch(fetchIsLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(fetchIsLoading(false));
      const userData = await response.json();
      const user = findUser(userData, email, password);
      dispatch(userFetchSuccess(user));
    } catch (e) {
      dispatch(fetchErrored(true));
    }
  };
};

export const findUser = (userData, email, password) => {
  return userData.data.find(user => user.email === email && user.password === password);
};