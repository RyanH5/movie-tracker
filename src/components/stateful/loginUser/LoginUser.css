input {
  outline: none;
  border: none; }

.login-user-container {
  width: 100%;
  height: calc(100vh - 160px);
  justify-content: center;
  align-items: center; }
  .login-user-container form {
    margin: 0;
    padding: 0;
    width: 560px; }
    .login-user-container form p {
      color: #fff;
      text-align: center; }
    .login-user-container form span {
      width: 100%;
      display: block;
      font-size: 30px;
      color: #fefefe;
      line-height: 1.2;
      text-transform: uppercase;
      text-align: center;
      padding-bottom: 30px; }
    .login-user-container form .wrap-input {
      display: flex;
      flex-wrap: wrap;
      align-items: center;
      width: 100%;
      height: auto;
      position: relative;
      border: 1px solid #e0e0e0;
      border-bottom: none;
      background-color: #fff;
      float: left; }
      .login-user-container form .wrap-input input {
        margin: 1rem 0;
        height: 100%;
        font-size: 15px;
        color: #555555;
        line-height: 1.2;
        display: block;
        width: 100%;
        background: transparent;
        padding: 0 30px; }
      .login-user-container form .wrap-input.top {
        border-top-left-radius: 12px;
        border-top-right-radius: 12px; }
      .login-user-container form .wrap-input.bottom {
        width: 100%; }
    .login-user-container form button {
      font-size: 1rem;
      color: #fff;
      line-height: 1.2;
      text-transform: uppercase;
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 0 20px;
      width: 100%;
      height: 60px;
      border-bottom-left-radius: 12px;
      border-bottom-right-radius: 12px;
      overflow: hidden;
      background: #111111;
      position: relative;
      z-index: 1; }
      .login-user-container form button:hover {
        background: #BF2604;
        background: -webkit-linear-gradient(right, #F2790F, #BF2604);
        background: -o-linear-gradient(right, #F2790F, #BF2604);
        background: -moz-linear-gradient(right, #F2790F, #BF2604);
        background: linear-gradient(right, #F2790F, #BF2604);
        -webkit-transition: all 0.4s;
        -o-transition: all 0.4s;
        -moz-transition: all 0.4s;
        transition: all 0.4s;
        cursor: pointer; }

/*# sourceMappingURL=LoginUser.css.map */
