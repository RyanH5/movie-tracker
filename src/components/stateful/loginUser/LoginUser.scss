
input {
  outline: none;
  border: none;
}

.login-user-container {
  width: 100%;
  height: 100vh;
  justify-content: center;
  align-items: center;
  form {
    margin: 0;
    padding: 0;
    width: 560px;
    p {
      color: #fff;
      text-align: center;
    }
    span {
      width: 100%;
      display: block;
      font-size: 30px;
      color: #fefefe;
      line-height: 1.2;
      text-transform: uppercase;
      text-align: center;
      padding-bottom: 30px;
    }
    .wrap-input {
      display: flex;
      flex-wrap: wrap;
      align-items: center;
      width: 100%;
      height: auto;
      position: relative;
      border: 1px solid #e0e0e0;
      border-bottom: none;
      background-color: #fff;
      float: left;
      input {
        margin: 1rem 0;
        height: 100%;
        font-size: 15px;
        color: #555555;
        line-height: 1.2;
        display: block;
        width: 100%;
        background: transparent;
        padding: 0 30px;
      }
      &.top {
        border-top-left-radius: 12px;
        border-top-right-radius: 12px;
      }
      &.bottom {
        width: 100%;
      }
    }
    button {
      font-size: 1rem;
      color: #fff;
      line-height: 1.2;
      text-transform: uppercase;
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 0 20px;
      width: 100%;
      height: 60px;
      border-bottom-left-radius: 12px;
      border-bottom-right-radius: 12px;
      overflow: hidden;
      background: #111111;
      position: relative;
      z-index: 1;
      &:hover {
        background: #BF2604;
        background: -webkit-linear-gradient(to left, #F2790F, #BF2604);
        background: -o-linear-gradient(to left, #F2790F, #BF2604);
        background: -moz-linear-gradient(to left, #F2790F, #BF2604);
        background: linear-gradient(to left, #F2790F, #BF2604);
        -webkit-transition: all 0.4s;
        -o-transition: all 0.4s;
        -moz-transition: all 0.4s;
        transition: all 0.4s;
        cursor: pointer;
      }
    }
  }
}