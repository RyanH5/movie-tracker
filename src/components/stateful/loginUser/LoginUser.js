import React, { Component } from 'react';
import { fetchDatabase } from '../../../actions/loginActions';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchFavorites } from '../api/Api';

import './LoginUser.css';
import { captureFavorites } from '../../../actions/movieActions/movieActions';

export class LoginUser extends Component {
  constructor () {
    super();
    this.state = {
      password: '',
      email: ''
    };
  }

  handleChange = (event) => {
    const {name, value} = event.target;
    this.setState({[name]: value});
  };

  handleSubmit = async (event) => {
    event.preventDefault();
    const url = 'http://localhost:3000/api/users';
    await this.props.fetchDatabase(url, this.state.email, this.state.password);
    if (this.props.loginSuccess) {
      const favorites = await fetchFavorites(this.props.loginSuccess.userId);
      this.props.captureFavorites(favorites);
      this.props.history.push('/');
    }
  };

  render () {
    return (
      <div className="login-user-container d-flex">
        <div className="login-user-form-wrapper">
          <form onSubmit={this.handleSubmit}>
            <span className="">
						Login User
					  </span>
            {this.props.loginErrored &&
            <p>Email and password do not match</p>
            }
            <div className="wrap-input top">
              <input
                type="text"
                name="email"
                value={this.state.email}
                placeholder="ex. example@gmail.com"
                onChange={this.handleChange}
              />
            </div>
            <div className="wrap-input">
              <input
                type="text"
                name="password"
                value={this.state.password}
                placeholder="password"
                onChange={this.handleChange}
              />
            </div>
            <button>LOG-IN</button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  loginSuccess: state.loginSuccess,
  loginErrored: state.loginErrored,
  loginLoading: state.loginLoading
});

const mapDispatchToProps = (dispatch) => ({
  fetchDatabase: (url, email, password) => dispatch(fetchDatabase(url, email, password)),
  captureFavorites: (favorites) => dispatch(captureFavorites(favorites))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(LoginUser));
