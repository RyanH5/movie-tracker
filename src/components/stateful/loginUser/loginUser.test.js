import React from 'react';
import { LoginUser, mapStateToProps, mapDispatchToProps, handleChange } from './LoginUser';
import { shallow, mount } from 'enzyme';
import { fetchDatabase } from '../../../actions/loginActions';

describe('LoginUser', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LoginUser />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct default state', () => {
    const expected = {
      password: '',
      email: ''
    };

    expect(wrapper.state()).toEqual(expected);
  });

  it('should have an initial state', () => {
    expect(wrapper.state().password).toBe('');
    expect(wrapper.state().email).toEqual('');
  });

  describe('handleSubmit', () => {
    let mockUrl;
    let mockData;
    let mockEmail;
    let mockPassword;

    beforeEach(() => {
      mockUrl = 'mockurl.com';
      mockEmail= 'ok@mgmail.com';
      mockPassword = '12345';
      mockData = {
        results:
          [{status:"success", data:[{id:1, name:"Taylor", password:"password", email:"tman2272@aol.com"}]
          }]
      }

      window.fetch = jest.fn().mockImplementation(() => {
        return Promise.resolve({
          status: 200,
          json: () => Promise.resolve(mockData)
        });
      });
    });

    it('should call fetch with the correct params', () => {
      const mockFetch = jest.fn()
      fetchDatabase(mockUrl, mockEmail, mockPassword);
      expect(mockFetch).toHaveBeenCalledWith(mockUrl, mockEmail, mockPassword);
    });
  });
});
