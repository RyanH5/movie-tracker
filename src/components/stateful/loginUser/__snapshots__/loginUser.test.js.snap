// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginUser should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-user-form-wrapper"
>
        <form
                onSubmit={[Function]}
        >
                <span
                        className=""
                >
                        Login User
                </span>
                <div
                        className="wrap-input top"
                >
                        <input
                                name="email"
                                onChange={[Function]}
                                placeholder="ex. example@gmail.com"
                                type="text"
                                value=""
                        />
                </div>
                <div
                        className="wrap-input"
                >
                        <input
                                name="password"
                                onChange={[Function]}
                                placeholder="password"
                                type="text"
                                value=""
                        />
                </div>
                <button>
                        LOG-IN
                </button>
        </form>
</div>,
      "className": "login-user-container d-flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <span
                    className=""
          >
                    Login User
          </span>
          <div
                    className="wrap-input top"
          >
                    <input
                              name="email"
                              onChange={[Function]}
                              placeholder="ex. example@gmail.com"
                              type="text"
                              value=""
                    />
          </div>
          <div
                    className="wrap-input"
          >
                    <input
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              type="text"
                              value=""
                    />
          </div>
          <button>
                    LOG-IN
          </button>
</form>,
        "className": "login-user-form-wrapper",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className=""
>
              Login User
</span>,
            undefined,
            <div
              className="wrap-input top"
>
              <input
                            name="email"
                            onChange={[Function]}
                            placeholder="ex. example@gmail.com"
                            type="text"
                            value=""
              />
</div>,
            <div
              className="wrap-input"
>
              <input
                            name="password"
                            onChange={[Function]}
                            placeholder="password"
                            type="text"
                            value=""
              />
</div>,
            <button>
              LOG-IN
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login User",
              "className": "",
            },
            "ref": null,
            "rendered": "Login User",
            "type": "span",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="email"
                onChange={[Function]}
                placeholder="ex. example@gmail.com"
                type="text"
                value=""
/>,
              "className": "wrap-input top",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "ex. example@gmail.com",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="text"
                value=""
/>,
              "className": "wrap-input",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LOG-IN",
            },
            "ref": null,
            "rendered": "LOG-IN",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-user-form-wrapper"
>
          <form
                    onSubmit={[Function]}
          >
                    <span
                              className=""
                    >
                              Login User
                    </span>
                    <div
                              className="wrap-input top"
                    >
                              <input
                                        name="email"
                                        onChange={[Function]}
                                        placeholder="ex. example@gmail.com"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <div
                              className="wrap-input"
                    >
                              <input
                                        name="password"
                                        onChange={[Function]}
                                        placeholder="password"
                                        type="text"
                                        value=""
                              />
                    </div>
                    <button>
                              LOG-IN
                    </button>
          </form>
</div>,
        "className": "login-user-container d-flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
>
            <span
                        className=""
            >
                        Login User
            </span>
            <div
                        className="wrap-input top"
            >
                        <input
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="ex. example@gmail.com"
                                    type="text"
                                    value=""
                        />
            </div>
            <div
                        className="wrap-input"
            >
                        <input
                                    name="password"
                                    onChange={[Function]}
                                    placeholder="password"
                                    type="text"
                                    value=""
                        />
            </div>
            <button>
                        LOG-IN
            </button>
</form>,
          "className": "login-user-form-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className=""
>
                Login User
</span>,
              undefined,
              <div
                className="wrap-input top"
>
                <input
                                name="email"
                                onChange={[Function]}
                                placeholder="ex. example@gmail.com"
                                type="text"
                                value=""
                />
</div>,
              <div
                className="wrap-input"
>
                <input
                                name="password"
                                onChange={[Function]}
                                placeholder="password"
                                type="text"
                                value=""
                />
</div>,
              <button>
                LOG-IN
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login User",
                "className": "",
              },
              "ref": null,
              "rendered": "Login User",
              "type": "span",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="email"
                  onChange={[Function]}
                  placeholder="ex. example@gmail.com"
                  type="text"
                  value=""
/>,
                "className": "wrap-input top",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "placeholder": "ex. example@gmail.com",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  name="password"
                  onChange={[Function]}
                  placeholder="password"
                  type="text"
                  value=""
/>,
                "className": "wrap-input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "placeholder": "password",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LOG-IN",
              },
              "ref": null,
              "rendered": "LOG-IN",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
